logger.go

```diff
@@ -199,36 +147,71 @@ func (l *Logger) With(keyValues ...interface{}) telemetry.Logger {
 // Context attaches provided Context to the Logger allowing metadata found in
 // this context to be used for log lines and metrics labels.
 func (l *Logger) Context(ctx context.Context) telemetry.Logger {
-	newLogger := &Logger{
-		args:        make([]interface{}, len(l.args)),
-		ctx:         ctx,
-		metric:      l.metric,
-		level:       atomic.LoadInt32(&l.level),
-		writer:      l.writer,
-		name:        l.name,
-		description: l.description,
-	}
-
-	copy(newLogger.args, l.args)
-
+	newLogger := l.clone()
+	newLogger.ctx = ctx
 	return newLogger
 }
 
 // Metric attaches provided Metric to the Logger allowing this metric to
 // record each invocation of Info and Error log lines. If context is available
 // in the logger, it can be used for Metrics labels.
 func (l *Logger) Metric(m telemetry.Metric) telemetry.Logger {
+	newLogger := l.clone()
+	newLogger.metric = m
+	return newLogger
+}
+
+// clone the current logger and return it
+func (l *Logger) clone() *Logger {
 	newLogger := &Logger{
 		args:        make([]interface{}, len(l.args)),
 		ctx:         l.ctx,
-		metric:      m,
+		metric:      l.metric,
 		level:       atomic.LoadInt32(&l.level),
```

Comment:
This looks like it might be a problem... once you clone a logger (because you add Context or key-value pairs), you lose access to adjusting the log level from the original logger or scoped logger.

Reply:
Fixed